app/models/camp_model.php

```diff
@@ -0,0 +1,136 @@
+                return false;
+            }
+        }
+        if (!$stmt->execute()) {
+            return false;
+        }
+        $result = $stmt->get_result();
+        if ($result) {
+            return $result;
+        }
+        return $stmt;
+    }
+
+    function add_new($name, $organisation_id, $population, $volunteers, $status, $fname, $lname, $dob, $gender,
+     $ch_phone_no, $ch_email, $ch_mailing_list, $ch_mailing_address, $ch_gis_url, 
+     $phone_no, $email, $mailing_list, $mailing_address, $gis_url) {
+        $this->DB->autocommit(false);
+        //camp head contact details
+        
+        $camp_head_cd = $this->execute(
+            'INSERT INTO `contact_details`(`phone_no`, `email`, `mailing_list`, `mailing_address`, `gis_url`) VALUES (?, ?, ?, ?, ?)',
+            'sssss',
+            array(
+                &$ch_phone_no,
+                &$ch_email,
+                &$ch_mailing_list,
+                &$ch_mailing_address,
+                &$ch_gis_url,
+            )
+        );
+
+        if ($camp_head_cd) {
+            $camp_head_cd = $camp_head_cd->insert_id;
+        }
+        else {
+            return false;
+        }
+
+        //camp head id
+
+        $camp_head = $this->execute(
+            'INSERT INTO `persons`(`fname`, `lname`, `gender`, `dob`, `contact_id`) VALUES (?, ?, ?, ?, ?)',
+            'ssssi',
+            array(
+                &$fname,
+                &$lname,
+                &$gender,
+                &$dob,
+                &$camp_head_cd,
+            )
+        );
+
+        if ($camp_head) {
+            $camp_head = $camp_head->insert_id;
+        }
+        else {
+            return false;
+        }
+
+        $contact_id = $this->execute(
+            'INSERT INTO `contact_details`(`phone_no`, `email`, `mailing_list`, `mailing_address`, `gis_url`) VALUES (?, ?, ?, ?, ?)',
+            'sssss',
+            array(
+                &$phone_no,
+                &$email,
+                &$mailing_list,
+                &$mailing_address,
+                &$gis_url,
+            )
+        );
+
+        if ($contact_id) {
+            $contact_id = $contact_id->insert_id;
+        }
+        else {
+            return false;
+        }
+
+
+
+        $camp_id = $this->execute(
+            'INSERT INTO `camps`(`name`,`organisation_id`,`camp_head`,`population`,`volunteers`,`status`,`contact_id`) VALUES (?, ?, ? , ? , ? , ? , ?)',
+            'siisssi',
+            array(
+                &$name,
+                &$organisation_id,
+                &$camp_head,
+                &$population,
+                &$volunteers,
+                &$status,
+                &$contact_id
+            )
+        );
+       
+        if ($camp_id) {
+            $camp_id = $camp_id->insert_id;
+        }
+        else {
+            return false;
+        }
+
+        return $this->DB->commit();
+    }
+    function get_camps() {
```

Comment:
If we don't have view functionality, this is also not needed.

Or just add view for list, too? With the old version of controller.